airflow/contrib/hooks/hortonworks_ambari_hook.py

```diff
@@ -0,0 +1,131 @@
+
+        :param datas: parametres of Hive script
+        :type datas: object
+
+        """
+
+        self.method = "POST"
+        hive_endpoint = "templeton/v1/"
+        statement_non_terminated_status_list = ['INIT', 'RUNNING', 'PREP']
+        submit_endpoint = hive_endpoint + "hive"
+
+        self.log.debug("Submiting hive  Script: %s", str(datas))
+        response = self.run(endpoint=submit_endpoint, data=datas, headers=self.headers)
+
+        job_id = response["id"]
+        status_endpoint = hive_endpoint + "jobs/" + str(job_id)
+        self.log.info("Finished submitting hive script job_id: %s", job_id)
+
+        statements_state = 'INIT'
+        while statements_state in statement_non_terminated_status_list:
+
+            # todo: test execution_timeout
+            time.sleep(5)
+            try:
+                result = self._get_job_status(endpoint=status_endpoint)["status"]["state"]
+            except Exception:
+                pass
+
+            statements_state = result["status"]["state"]
+            statements_exit_value = result["exitValue"]
+            if statements_state is not None:
+                # Check submitted job's result
+                if statements_state in ['KILLED', 'FAILED'] or statements_exit_value != "0":
+                    result = "job failed. state: %s", statements_state
+                    self.log.error(result)
+                    raise AirflowException(result)
+                else:
+                    self.log.debug("Checking Hive job: %s", statements_state)
+
+        self.log.debug("Statement %s ", statements_state)
+        self.log.debug("Finished executing WebHCatHiveSubmitOperator")
+
+    def submit_spark_job(self, datas):
+        """
+        Submit spark job or spark-sql to Azure HDInsight Cluster
+
+        :param datas: parametres of Spark Job
+        :type datas: object
+        """
+
+        statement_non_terminated_status_list = ['INIT', 'starting', 'running', "waiting", "available"]
+        spark_endpoint = "livy/batches"
+        submit_endpoint = spark_endpoint
+        self.method = "POST"
+        datas["user.name"] = self.username
+
+        self.log.info("Submiting spark Job: %s ", json.dumps(datas))
+        response = self.run(endpoint=submit_endpoint, data=json.dumps(datas))
+
+        job_id = response["id"]
+        self.log.debug("Finished submitting spark  job_id: %s", str(job_id))
+
+        statements_state = 'INIT'
+        while statements_state in statement_non_terminated_status_list:
+
+            # todo: test execution_timeout
+            time.sleep(5)
+            try:
+                status_endpoint = spark_endpoint + "/" + str(job_id)
+                statements_state = self._get_job_status(endpoint=status_endpoint)["state"]
+            except Exception:
+                pass
+            # Check submitted job's status
+            if statements_state in ['error', 'cancelling', 'cancelled']:
```

Comment:
What about state 'dead'?

Reply:
Actually, it can behavior like Airflow sensor, it's very simple. Maybe next feature, after removing this loop, I can write  airflow azure sensor like AWS step sensor.

Reply:
In this case, it will pass to any  exceptions for  checking job result state. If I could throw exception, airflow think like job fail, but only  job result checking failing. It's not error.

Reply:
From what I've seen, the behavior of a Spark application that crashes as it starts is "starting, running, dead". In this case wouldn't you want your Airflow task to fail?